src/MemObject.h

```diff
@@ -167,6 +167,13 @@ class MemObject
 
     SwapOut swapout;
 
+    /// whether the object is being updated on disk
+    // TODO: expand the scope to common swapouts as well
+    Lock swapoutInterest;
```

Comment:
I checked separately that sizeof(swapoutInterest) is 16 (vs 4, if without virtual destructor), i.e., it keeps vtable pointer. That means that we should consider moving to a boolean variable instead (implying other related problems fixed first).